### ISSUE TITLE
SyncBatchNorm doesn't work when I set track_running_stats False

### ISSUE THREAD

**Issue by RERV** [pytorch/pytorch#40147]
Hi, 
I try to apply a DDP to a model. However, when I set track_running_stats False, it output a error
![image](https://user-images.githubusercontent.com/39251602/84854955-8834d180-b095-11ea-9725-723310f81b50.png)
And when I turn it to True, it works well. Then I read the source
![image](https://user-images.githubusercontent.com/39251602/84855266-42c4d400-b096-11ea-9aef-8d9a8e3ed858.png)

and try to print it, it says
![image](https://user-images.githubusercontent.com/39251602/84855244-350f4e80-b096-11ea-82c9-daae77f56a3f.png)

running mean and running var is none. I konw it beacause I set track_running_stas False. 
And the actual bug is in here
![image](https://user-images.githubusercontent.com/39251602/84855389-9a633f80-b096-11ea-8169-b36a571d475a.png)
![image](https://user-images.githubusercontent.com/39251602/84855404-a222e400-b096-11ea-9fbf-93b848f561a7.png)
It seems that the "None" of running mean and the running var caused it. But I did need it set to False. I would be grateful if someone else could tell me how to fix it.

**Comment by RERV** [comment]
I run it on a single node with 2 gpu,
![image](https://user-images.githubusercontent.com/39251602/84855713-5886c900-b097-11ea-8085-58914278d268.png)

And the output of the print I said is here.
![image](https://user-images.githubusercontent.com/39251602/84855844-b9160600-b097-11ea-9787-3595d3cbb8e7.png)
（Forgive me for the wrong spelling）

The syncbn used here
![image](https://user-images.githubusercontent.com/39251602/84855899-dd71e280-b097-11ea-8341-3b728fbc1a42.png)

**Comment by rohan-varma** [comment]
Hi @RERV, thanks for reporting this issue! Could you please provide a script to reproduce the bug you're seeing and the PT version (1.4, 1.5, nightly, master etc) you're using? Also - would be great if you can post these as plaintext instead of an image, so it's easier to read and copy/paste.

**Comment by RERV** [comment]
> Hi @RERV, thanks for reporting this issue! Could you please provide a script to reproduce the bug you're seeing and the PT version (1.4, 1.5, nightly, master etc) you're using? Also - would be great if you can post these as plaintext instead of an image, so it's easier to read and copy/paste.


`import argparse
import numpy as np
import torch
import torch.nn as nn
import torch.optim as optim
from torch.utils import data


parser = argparse.ArgumentParser()
parser.add_argument('--local_rank', type=int, default=0)


class Dataset(data.Dataset):
    def __init__(self):
        super().__init__()
        self.data = torch.randn(1000, 32, 16, 16)

    def __len__(self):
        return len(self.data)

    def __getitem__(self, item):
        return self.data[item]


def main():
    torch.manual_seed(0)
    np.random.seed(0)
    global args
    args = parser.parse_args()

    global cuda
    print("Args local rank is {} type is {}".format(args.local_rank, type(args.local_rank)))
    torch.cuda.set_device(args.local_rank)
    torch.distributed.init_process_group(
        backend='nccl',
        init_method='env://'
    )
    cuda = torch.device(f'cuda:{args.local_rank}')


    # The bug is here !!!!!!!!!!!!!!
    # If I turned track_running_stats false, it didn't work
    # And I turned it to true, it works well
    model = nn.Sequential(
        nn.Conv2d(32, 16, 3, padding=1),
        nn.BatchNorm2d(16, track_running_stats=False),
        nn.ReLU()
    )
    model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
    model.to(cuda)
    model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[args.local_rank])

    criterion = nn.MSELoss().to(cuda)

    params = model.parameters()
    optimizer = optim.Adam(params, lr=0.001)

    dataset = Dataset()
    sampler = torch.utils.data.distributed.DistributedSampler(dataset)

    data_loader = data.DataLoader(dataset,
                                  batch_size=4,
                                  sampler=sampler,
                                  shuffle=False,
                                  num_workers=32,
                                  pin_memory=True,
                                  drop_last=True)




    EPOCH = 3
    for epoch in range(EPOCH):
        model.train()
        for idx, item in enumerate(data_loader):
            item = item.to(cuda)
            # For simple, we learn a identical function
            old_item = item[:, :16, :, :]
            item = model(item)
            loss = criterion(item, old_item)
            optimizer.zero_grad()
            loss.backward()
            optimizer.step()


if __name__ == '__main__':
    main()

`

I wrote a sample example, you can run it with 

> `CUDA_VISIBLE_DEVICES=0,1 python -m torch.distributed.launch --nproc_per_node=2 bug_test.py `
And you can change batchnorm2d track_running_stats to see the differences.

Thanks!

**Comment by RERV** [comment]
pytorch is 1.5.0

**Comment by svenstaro** [comment]
What am I supposed to do here? Why the highlight?

**Comment by RERV** [comment]
@rohan-varma hello, have you found out what caused it?
thanks.

**Comment by cdzhan** [comment]
@RERV It is fixed in newer version of pytorch, such as pytorch 1.9. You can have a try.
